miio/gateway.py

```diff
@@ -928,6 +1001,67 @@ def update(self):
         self._props.status_ch1 = values[1]
         self._props.load_power = values[2]
 
+class Cube(SubDevice):
+    """Subdevice Cube specific properties and methods"""
+
+    properties = []
+
+    @command()
+    def install_move_script(self, encoded_token):
```

Comment:
All these installs should probably be just a single method in `SubDevice`: `install_event_handler` or something like that, that would take 1) device, 2) event, 3) event receiver, and 4) method to be called.

The subdevices themselves should have a class attribute defining the available events. So in the case of the cube, the Cube class should define that it has events:
```
@property
def events():
    return ["tap_twice", "shake"]
```
The users of the lib can then request available events, and create a script for those.